web/src/hash_util.ts

```diff
@@ -127,56 +132,56 @@ export function operators_to_hash(operators) {
     return hash;
 }
 
-export function by_sender_url(reply_to) {
+export function by_sender_url(reply_to: string): string {
     return operators_to_hash([{operator: "sender", operand: reply_to}]);
 }
 
-export function pm_with_url(reply_to) {
+export function pm_with_url(reply_to: string): string {
     const slug = people.emails_to_slug(reply_to);
     return "#narrow/dm/" + slug;
 }
 
-export function huddle_with_url(user_ids_string) {
+export function huddle_with_url(user_ids_string: string): string {
     // This method is convenient for callers
     // that have already converted emails to a comma-delimited
     // list of user_ids.  We should be careful to keep this
     // consistent with hash_util.decode_operand.
     return "#narrow/dm/" + user_ids_string + "-group";
 }
 
-export function by_conversation_and_time_url(message) {
+export function by_conversation_and_time_url(message: Message): string {
     const absolute_url =
         window.location.protocol +
         "//" +
         window.location.host +
         "/" +
         window.location.pathname.split("/")[1];
 
-    const suffix = "/near/" + internal_url.encodeHashComponent(message.id);
+    const suffix = "/near/" + internal_url.encodeHashComponent(message.id.toString());
 
     if (message.type === "stream") {
-        return absolute_url + by_stream_topic_url(message.stream_id, message.topic) + suffix;
+        return absolute_url + by_stream_topic_url(message.stream_id!, message.topic) + suffix;
```

Comment:
I guess this `!` is a case where our type for `Message` isn't precise enough for TypeScript to infer that this must exist if it's a stream message. 

Can be a good follow-up to look into.